hopp/tools/design/wind/floris_helper_tools.py

```diff
@@ -0,0 +1,124 @@
+import os
+import numpy as np
+from hopp.utilities.utilities import write_yaml
+from floris.turbine_library.turbine_previewer import INTERNAL_LIBRARY
+from hopp.utilities import load_yaml
+from hopp.tools.design.wind.turbine_library_interface_tools import get_floris_turbine_specs
+from hopp.tools.design.wind.turbine_library_tools import check_turbine_name, check_turbine_library_for_turbine
+
+def check_output_formatting(orig_dict):
+    """Recursive method to convert arrays to lists and numerical entries to floats.
+        This is primarily used before writing a dictionary to a .yaml file to ensure
+        proper output formatting.
+
+    Args:
+        orig_dict (dict): input dictionary
+
+    Returns:
+        dict: input dictionary with reformatted values.
+    """
+    for key, val in orig_dict.items():
+        if isinstance(val, dict):
+            tmp = check_output_formatting(orig_dict.get(key, { }))
+            orig_dict[key] = tmp
+        else:
+            if isinstance(key, list):
+                for i,k in enumerate(key):
+                    if isinstance(orig_dict[k], str):
+                        orig_dict[k] = (orig_dict.get(key, []) + val[i])
+                    elif isinstance(orig_dict[k], bool):
+                        orig_dict[k] = (orig_dict.get(key, []) + val[i])
+                    elif isinstance(orig_dict[k], (list, np.ndarray)):
+                        new_val = [float(v) for v in val[i]]
+                        orig_dict[k] = new_val
+                    else:
+                        orig_dict[k] = float(val[i])
+            elif isinstance(key,str):
+                if not isinstance(orig_dict[key], (str, bool)):
+                    if isinstance(orig_dict[key], (list, np.ndarray)):
+                        new_val = [float(v) for v in val]
+                        orig_dict[key] = new_val
+                    else:
+                        orig_dict[key] = float(val)
+    return orig_dict
+
+def write_floris_layout_to_file(layout_x,layout_y,output_dir,turbine_desc):
+    """Export wind farm layout to floris-friendly .yaml file.
+
+    Args:
+        layout_x (List[float]): x-coordinates of turbines
+        layout_y (List[float]): y-coordinates of turbines
+        output_dir (str): output folder to write layout file to.
+        turbine_desc (str): turbine name or description.
+    """
+
+    layout_x = [float(x) for x in layout_x]
+    layout_y = [float(y) for y in layout_y]
+
+    layout = {"layout_x":layout_x,"layout_y":layout_y}
+    n_turbs = len(layout_x)
+    output_fpath = os.path.join(output_dir,f"layout_{turbine_desc}_{n_turbs}turbs.yaml")
+    write_yaml(output_fpath,layout)
+
+def write_turbine_to_floris_file(turbine_dict,output_dir):
+    """Export turbine model to floris-friendly .yaml file.
```

Comment:
A minor nitpick, but can you change the reference to YAML files from `.yaml files` to `YAML files`?